src/main/java/net/dv8tion/jda/core/entities/Message.java

```diff
@@ -1254,4 +1262,247 @@ public boolean isImage()
+         *         </ul>
+         */
+        public Activity(ActivityType type, String partyId, Application application)
+        {
+            this.type = type;
+            this.partyId = partyId;
+            if ((type == ActivityType.GAME && application == null) || (type != ActivityType.GAME && application != null))
+                throw new IllegalArgumentException("Either the ActivityType is wrong or the Application is null!");
+            this.application = application;
+        }
+
+        /**
+         *
+         * @return the type of the activity.
+         */
+        public ActivityType getType()
+        {
+            return type;
+        }
+
+        /**
+         * A {@link java.lang.String String} probably containing the activity service and its host (a {@link net.dv8tion.jda.core.entities.User User}) id.
+         * <br>For example : "spotify:86699011792191488".
+         *
+         * @return the partys id.
+         */
+        public String getPartyId()
+        {
+            return partyId;
+        }
+
+        /**
+         * Shortcut for grabbing the service of the activity if the {@link net.dv8tion.jda.core.entities.Message.ActivityType ActivityType} equals
+         * {@link net.dv8tion.jda.core.entities.Message.ActivityType#PARTY ActivityType.PARTY}.
+         *
+         * @return the service name like "spotify".
+         */
+        public String getService()
+        {
+            return (this.type == ActivityType.PARTY) ? partyId.split(":", 2)[0] : null;
+        }
+
+        /**
+         * This is a shortcut for getting the host id of the activity if the {@link net.dv8tion.jda.core.entities.Message.ActivityType ActivityType} equals
+         * {@link net.dv8tion.jda.core.entities.Message.ActivityType#PARTY ActivityType.PARTY}.
+         *
+         * @return the hosts id like "86699011792191488".
+         */
+        public String getHostId()
+        {
+            return (this.type == ActivityType.PARTY) ? partyId.split(":", 2)[1] : "-1";
+        }
+
+        /**
+         * Wrapping method for getting the {@link net.dv8tion.jda.core.entities.Message.Activity#getHostId()} as long.
+         *
+         * @return the hosts id.
+         *
+         * @see net.dv8tion.jda.core.entities.Message.Activity#getHostId()
+         */
+        public long getHostIdLong()
+        {
+            return Long.parseLong(this.getHostId());
+        }
+
+        /**
+         *
+         * @return A possibly {@code null} {@link net.dv8tion.jda.core.entities.Message.Application}.
+         */
+        public Application getApplication()
+        {
+            return application;
+        }
+
+        /**
+         *
+         * @return {@link org.json.JSONObject} that can be used for sending an {@link net.dv8tion.jda.core.entities.Message.Activity Activity} in a {@link net.dv8tion.jda.core.entities.Message}.
+         */
+        public JSONObject toJSONObject()
```

Comment:
There is only one place where this is used, if I'm not mistaken.
It makes more sense to just do this in `MessageAction` directly.

Reply:
OK will move it!